cogs/hosts.py

```diff
@@ -154,51 +163,72 @@ async def temphost(self, ctx):
-        calendar_today = "{:02d}".format(now.year) + "{:02d}".format(now.month) + "{:02d}".format(now.day)
-        query = "SELECT thread_id FROM xf_andy_calendar WHERE calendar_date = :field_value"
-        values = {"field_value": calendar_today}
-        results = await database.fetch_all(query=query, values=values)
-
-        for result in results:
-            #check if the users forum account is linked to that thread in the calendar thread
-            try:
-                #select user_id from xf_thread where thread_id='49093'
-                query = "select user_id from xf_thread where thread_id = :field_value"
-                values = {"field_value": result[0]}
-                thread_creator_id = await database.fetch_one(query=query, values=values)
-
-            except:
-                print("Can't find the user who created the thread. This is a rare error only see in testing.")
-                await ctx.edit(content=f"{error} Can't find user who created the thread on the calendar. You'll need to be added manually. This is a rare error only see in testing.")
-                return
-            
-            #check if the thread found matches the user_id linked
-            if thread_creator_id[0] != user_forum_id:
-                print("User ID on thread does not match the discord ID linked on the forums.")
-                await ctx.edit(content=f"{error} You don't seem to have an event thread on the calendar or you aren't the one who created the thread!")
-                return
-
-            #add to role
-            user_secondary_groups = await self.get_user_roles(thread_creator_id[0])
-
-            if 43 in user_secondary_groups:
-                await ctx.edit(content=f"{warning} {ctx.author.name} already has the temp-hosts group on the forum!")
-                time.sleep(3)
-            else:
-                message = await ctx.edit(content=f"{loading} Adding {ctx.author.name} to temp-hosts group on the forums...")
-                time.sleep(2)
-                await self.add_user_temp_host(ctx, message, user_forum_id, user_secondary_groups, ctx.author)
-            break
+        #check if server has a lot of people on it
+            #how to know what server they are on
+                #rcon
+            #is that worth the effort?
+
+        query = "SELECT provider_key FROM xf_user_connected_account WHERE provider = 'th_cap_steam' AND user_id = :user_id"
+        values = {"user_id": user_forum_id}
+        result = await database.fetch_one(query=query, values=values)
+        if result is None:
+            await ctx.respond(f"{error} You do not have a steam account linked.")
+            return
+        user_steam_id = result[0].decode("utf-8")
+
+        user_steam3 = SteamID(user_steam_id).as_steam3
+
+        on_server = False
+
+        response_us = await rcon('api_steamids', host="us.tf2maps.net", port=27015, passwd=sftp_config.us_tf2maps_net.rcon)
+        players_us = ast.literal_eval(response_us)
```

Comment:
you should never ever ever be doing AST manipulation. 

what is the rcon function returning that you cant parse here?